### PR TITLE
Common overhaul

### PR DESCRIPTION
**Pull request by mharsat**

### CODE REVIEW

File: .e2e.env.list [hunk 1]
```diff
@@ -1,6 +1,7 @@
 APPLITOOLS_API_KEY
 APPLITOOLS_BATCH_ID
 APPLITOOLS_DONT_CLOSE_BATCHES="true"
+APPLITOOLS_IS_DISABLED
```
Comment (shaulgo) [inline .e2e.env.list]: does this need to be set to false?

Comment (mharsat) [reply .e2e.env.list]: It was set in travis CI settings to `false` (and defaults to false anyway).
This line is needed so when we'll need to set the env variable to `true` in travis, the value will be passed into the docker and take effect

File: examples/main/config/webpack.common.js [hunk 2]
```diff
@@ -23,7 +23,12 @@ module.exports = env => ({
     extensions: ['.js', '.jsx', '.json'],
     symlinks: false,
     alias: {
-      'wix-rich-content-common': path.resolve(PATHS.monorepo_root, 'packages', 'common', 'web'),
```
Comment (shaulgo) [inline examples/main/config/webpack.common.js]: should this be removed?

File: examples/viewer-ssr/package.json [hunk 3]
```diff
@@ -52,7 +52,7 @@
     "wix-bootstrap-ng": "latest",
     "wix-express-csrf": "latest",
     "wix-express-require-https": "latest",
-    "wix-rich-content-common": "file:../../packages/common/web",
+    "wix-rich-content-editor-common": "file:../../packages/editor-common/web",
```
Comment (shaulgo) [inline examples/viewer-ssr/package.json]: The viewer-ssr shouldn't have editor-common. Why do we still need it?

Comment (mharsat) [reply examples/viewer-ssr/package.json]: We probably don't

File: packages/editor-common/web/src/Components/Checkbox.jsx [hunk 4]
```diff
@@ -1,10 +1,10 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import classnames from 'classnames';
-import { mergeStyles } from '../Utils/mergeStyles';
+import { mergeStyles } from 'wix-rich-content-common';
 import { CheckIcon } from '../Icons';
 import styles from '../../statics/styles/checkbox.scss';
-import generalStyles from '../../statics/styles/general.scss';
+import generalStyles from 'wix-rich-content-common/dist/statics/styles/general.scss';
```
Comment (shaulgo) [inline packages/editor-common/web/src/Components/Checkbox.jsx]: did you take a look if this are needed in the viewer?

Comment (shaulgo) [reply packages/editor-common/web/src/Components/Checkbox.jsx]: I presume that the css can be split as well. But we can leave it for now. See that everything is working. And then do it.

Comment (mharsat) [reply packages/editor-common/web/src/Components/Checkbox.jsx]: Yes, besides in the Checkbox, `general.scss` is used in the AccessibilityListener which is needed by the viewer

Comment (mharsat) [reply packages/editor-common/web/src/Components/Checkbox.jsx]: Sounds good

File: packages/editor-common/web/src/Utils/draftUtils.js [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+    entityKey
+  ).set('selectionAfter', oldSelection);
+
+  return EditorState.push(editorState, newContentState, 'apply-entity');
+}
+
+export const hasLinksInEditorBlock = (block, contentState) => {
+  try {
+    if (block.findEntityRanges) {
+      return !!getLinkRangesInBlock(block, contentState).length;
```
Comment (shaulgo) [inline packages/editor-common/web/src/Utils/draftUtils.js]: you can remove all the other lines in this function

File: packages/fullScreen/web/package.json [hunk 6]
```diff
@@ -51,6 +51,7 @@
   },
   "dependencies": {
     "react-images": "1.0.0",
+    "wix-rich-content-editor-common": "6.4.0",
```
Comment (shaulgo) [inline packages/fullScreen/web/package.json]: fullscreen is part of the viewer. It shouldn't have editor-common

File: packages/editor-common/web/src/Utils/draftUtils.js (hunk 5 above)
Comment (mharsat) [reply packages/editor-common/web/src/Utils/draftUtils.js]: correct thanks!

File: examples/viewer-ssr/package.json (hunk 3 above)
Comment (mharsat) [reply examples/viewer-ssr/package.json]: It's only use of editor-common is to import `RichContentModal`. I believe it isn't needed in viewer-ssr?

File: packages/plugin-button/web/src/components/settings-component.jsx [hunk 7]
```diff
@@ -1,6 +1,7 @@
 import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
-import { mergeStyles, TextInput, Checkbox, isValidUrl } from 'wix-rich-content-common';
+import { TextInput, Checkbox } from 'wix-rich-content-editor-common';
+import { isValidUrl, mergeStyles } from 'wix-rich-content-common';
```
[the remark this answers is not in the document]
Comment (mharsat) [reply packages/plugin-button/web/src/components/settings-component.jsx]: Yes, for example it is used by the HTML component both in viewer and editor

File: packages/plugin-html/web/src/toolbar/HtmlEditPanel.jsx [hunk 8]
```diff
@@ -1,15 +1,8 @@
...
-  RadioGroupHorizontal,
-  TextInput,
-  InputWithLabel,
-  isValidUrl,
-  startsWithHttps,
-  hasProtocol,
-  mergeStyles,
-} from 'wix-rich-content-common';
+import { RadioGroupHorizontal, TextInput, InputWithLabel } from 'wix-rich-content-editor-common';
+import { mergeStyles, isValidUrl, startsWithHttps, hasProtocol } from 'wix-rich-content-common';
```
[the remark this answers is not in the document]
Comment (mharsat) [reply packages/plugin-html/web/src/toolbar/HtmlEditPanel.jsx]: Correct, will be refactored in a new PR

File: packages/plugin-sound-cloud/web/src/toolbar/soundCloudURLInputModal.jsx [hunk 9]
```diff
@@ -2,13 +2,8 @@ import PropTypes from 'prop-types';
...
 import classNames from 'classnames';
-import {
-  mergeStyles,
-  isSoundCloudUrl,
-  SettingsPanelFooter,
-  TextInput,
-  CloseIcon,
-} from 'wix-rich-content-common';
+import { SettingsPanelFooter, TextInput, CloseIcon } from 'wix-rich-content-editor-common';
+import { isSoundCloudUrl, mergeStyles } from 'wix-rich-content-common';
```
[the remark this answers is not in the document]
Comment (mharsat) [reply packages/plugin-sound-cloud/web/src/toolbar/soundCloudURLInputModal.jsx]: Correct, will be refactored in a new PR

File: packages/plugin-text-color/web/src/color-scheme-utils.js [hunk 10]
```diff
@@ -1,4 +1,4 @@
-import { isHexColor, validate } from 'wix-rich-content-common';
+import { validate, isHexColor } from 'wix-rich-content-common';
```
[the remark this answers is not in the document]
Comment (mharsat) [reply packages/plugin-text-color/web/src/color-scheme-utils.js]: No. It is used by the highlight plugin in viewer

File: examples/main/config/webpack.common.js (hunk 2 above)
Comment (mharsat) [reply examples/main/config/webpack.common.js]: both common and editor-common are now used

File: packages/editor-common/web/src/Components/Checkbox.jsx (hunk 4 above)
Comment (mharsat) [reply packages/editor-common/web/src/Components/Checkbox.jsx]: general.scss will be split in a new PR

File: packages/editor-common/web/src/Utils/draftUtils.js (hunk 5 above)
Comment (shaulgo) [reply packages/editor-common/web/src/Utils/draftUtils.js]: we need only this `return !!getLinkRangesInBlock(block, contentState).length;`

**Pull request merged**